### ISSUE TITLE
[2017.1.0] Metal Self Build Script Inconsistency

### ISSUE THREAD

**Issue by ColonelPanicks** [alces-software/metalware#243]
When running `metal build self` it seems to incorrectly render the list of scripts to be used for building. Below are more details on the inconsistency:
```
# domain.yaml
files:
  setup:
    - /opt/alces/install/scripts/00-repos.sh
    - /opt/alces/install/scripts/03-ganglia.sh
    - local-script.sh
    - https://raw.githubusercontent.com/alces-software/metalware-default/master/files/remote-script.sh
    - non-existent-local-script.sh
    - http://example.com/non-existent-remote-script.sh

# self.yaml
files:
  setup:
    - setup/dns.sh
    - setup/xinetd.sh
    - setup/dhcp.sh
    - setup/http.sh
    - /opt/alces/install/scripts/03-ganglia.sh
```

The render command would lead me to believe that the files list is overwritten (note how `00-repos.sh` is not in the `self` output)
```
[root@localhost ~]# cat /tmp/test
<%= files.setup %>
[root@localhost ~]# metal render /tmp/test
warning: Unset template parameter: alces.answers.pri_network_defined
warning: Unset template parameter: alces.answers.mgt_network_defined
warning: Unset template parameter: alces.answers.ib_network_defined
warning: Unset template parameter: alces.answers.bmc_network_defined
["/opt/alces/install/scripts/00-repos.sh", "/opt/alces/install/scripts/03-ganglia.sh", "local-script.sh", "https://raw.githubusercontent.com/alces-software/metalware-default/master/files/remote-script.sh", "non-existent-local-script.sh", "http://example.com/non-existent-remote-script.sh"]
[root@localhost ~]# metal render /tmp/test  self
["setup/dns.sh", "setup/xinetd.sh", "setup/dhcp.sh", "setup/http.sh", "/opt/alces/install/scripts/03-ganglia.sh"]
```

However the rendered directory shows that it's actually concatenating the lists and rendering all of the scripts
```
[root@localhost ~]# ls /var/lib/metalware/rendered/self/setup/
00-repos.sh  03-ganglia.sh  dhcp.sh  dns.sh  http.sh  local-script.sh  xinetd.sh
```

Ideally the files lists would be overwritten when inheriting.

**Comment by WilliamMcCumstie** [comment]
This bug will be fixed as part of the greater build refactor and will not be addressed directly. The issue will be remain open until such time it is fixed.

**Comment by WilliamMcCumstie** [comment]
Previously this was the expected behaviour as metalware internally creates a `build_files` list that causes the other files to be rendered. The bug is actually the `files` list being wrong which has lead to the confusion.

However the specification for metalware has now changed in the upcomming `2017.2.0` release. The files will no longer be appended to the higher level. Instead the files list will be replaced.

The new specification has also been implemented in `2017.1.1`
